### PR TITLE
Fix errcheck warnings, add to build

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request cleans up our error checking and adds `errcheck` to the precommit hook and Travis build. I also added a `Makefile` so that the precommit hook and Travis can share the same command settings without copy-and-paste.

### CODE REVIEW

File: httpd/handler.go [hunk 1]
```diff
@@ -248,7 +248,7 @@ func (h *Handler) serveStatus(w http.ResponseWriter, r *http.Request) {
 	} else {
 		b, _ = json.Marshal(data)
 	}
-	w.Write(b)
+	_, _ = w.Write(b)
```
Comment (corylanou) [inline httpd/handler.go]: Should we be using ignore for some of this;

`errcheck -ignore '[rR]ead|[wW]rite' path/to/package`

I feel like this creates a lot of noise for no reason in a package like this.

I agree with the errcheck concept, but there is good reason you don't need to check some of these.

Comment (benbjohnson) [inline httpd/handler.go]: I don't think we can ignore errors based on name. The two times when I ignore returned errors are when I'm dealing with something like `bytes.Buffer` which cannot return an error and when I'm writing out to an `http.ResponseWriter` where there's not really much to be done if it's not successful.

Comment (corylanou) [inline httpd/handler.go]: I'm not sure I follow you.  Do you mean "can't" or "shouldn't"?  I haven't tried it, but it feels like based on the documentation we can pass it the handler.go file and tell it to ignore error checks for `w.Write`.

I think for sure I'm having a very allergic reaction to this :-).

**Comment by otoolep** [comment]
We need a Makefile? Both :-) and :-( at the same time.

**Comment by otoolep** [comment]
Instead of 'make' how about a simple shell script that is called from Travis? make is a great tool, but it would be nice if our dev process did not require it.

**Comment by benbjohnson** [comment]
Closing as there's not consensus for a good way to integrate `errcheck`.

**Comment by benbjohnson** [comment]
@otoolep I find Makefiles really nice for small tasks. It seems easier to navigate than to have a bunch of tiny bash scripts littered through the code base (IMHO).

**Pull request closed without merging**